pkg/newlog/cmd/newlogd.go

```diff
@@ -1578,7 +1578,8 @@ func getDevTop10Inputs() {
 func getPtypeTimestamp(timeStr string) *timestamp.Timestamp {
 	t, err := time.Parse(time.RFC3339, timeStr)
 	if err != nil {
-		log.Fatal(err)
+		log.Warn(err)
```

Comment:
Not a good idea to produce a log from the logger. This can lead to infinite loop if (for some reason) you again fail to parse something.

Reply:
I agree. But in this case, getPtypeTimestamp() is called only at one point in the code ([here](https://github.com/lf-edge/eve/blob/master/pkg/newlog/cmd/newlogd.go#L765)). The only situation that would lead to a infinite loop as you described is that if log.Warn() generates wrong timestamps, which should not happen. However, silently change the timestamp can lead to the following situation:

1) Months later nobody will remember or care about these changes
2) Then somebody needs to debug a very annoying bug
3) Let's say some logs have bad timestamps (for any reason): log messages will have timestamps with no warning at all; and these timestamps will not correspond to the reality (the exact time which events happened), it might confuse who is debugging the system
4) It might be very difficult to realize timestamps were fixed by newlogd

Although I think is too much kill newlogd only because of wrong timestamps, I think is still the best option if you really won't use log.Warn() here.

:warning: One important observation :warning: 
Since I removed the parsing of VM console messages, we shouldn't have this timestamp problem anymore, unless some container or other entity that we care about it starts to send wrong timestamps. So maybe it might make sense let log.Fatal()

@eriknordmark, @rouming, thoughts?

Reply:
I'd agree with @rouming on this one. Later we will forget about this and it can shoot us in the leg pretty bad. At this point you are relying on a 3rd party library to provide you correct time formatting. From two evils of infinite log generation and unknown missing logs I'd choose former :)

Reply:
> So maybe it might make sense let log.Fatal()

Personally I don't like an idea of failing the whole system if you can't parse something. If you can't open a file - yes, crash it; if you can't create a directory for your logs - yes, but if parsing failed - no. So I would return nil and let the caller decide what to do: replace time on now() or something else.

Reply:
I think you are right @rouming . I've changed my PR with a new approach:

- Do not use log.Warn() so we avoid any chance of recursive calls.
- Return the error to the caller instead of fail, so the caller can decide what to do.
- If the timestamp is invalid, we cannot trust in the time information, so return default (initial) Unix time instead of the current time. This can give a good indication to who is analyzing the logs that the timestamp is invalid

I've also rebased with master.